airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/CdcPostgresSourceTest.java

```diff
@@ -871,9 +871,10 @@ protected void verifyCheckpointStatesByRecords() throws Exception {
         .toListAndClose(secondBatchIterator);
     assertEquals(recordsToCreate, extractRecordMessages(dataFromSecondBatch).size());
     final List<AirbyteStateMessage> stateMessagesCDC = extractStateMessages(dataFromSecondBatch);
-    if (!isOnLegacyPostgres()) {
-      assertTrue(stateMessagesCDC.size() > 1, "Generated only the final state.");
-    }
+    // We expect only one cdc state message, as all the records are inserted in a single transaction.
+    // Since
+    // lsn_commit only increases with a new transaction, we expect only one state message.
```

Comment:
are we not expecting a checkpoint state message (after the transaction) and a final state message in this case?

Reply:
The one state message we expect is after the dbz engine is shut down. In this case, there is just one transaction so only one state message is expected. (that's the change we're trying to test by comparing against `lsn_commit` and not `lsn_proc`)